ui/src/AzureActionHandler.ts

```diff
@@ -0,0 +1,86 @@
+    public constructor(extensionContext: ExtensionContext, outputChannel: OutputChannel, telemetryReporter?: TelemetryReporter) {
+        this._extensionContext = extensionContext;
+        this._outputChannel = outputChannel;
+        this._telemetryReporter = telemetryReporter;
+    }
+
+    public registerCommand(commandId: string, callback: (...args: any[]) => any): void {
+        this.registerCommandWithCustomTelemetry(commandId, (_properties: TelemetryProperties, _measurements: TelemetryMeasurements, ...args: any[]) => callback(...args));
+    }
+
+    public registerCommandWithCustomTelemetry(commandId: string, callback: (properties: TelemetryProperties, measurements: TelemetryMeasurements, ...args: any[]) => any): void {
+        this._extensionContext.subscriptions.push(commands.registerCommand(commandId, this.wrapCallback(commandId, (trackTelemetry: () => void, properties: TelemetryProperties, measurements: TelemetryMeasurements, ...args: any[]) => {
+            trackTelemetry(); // Always track telemetry for commands
+            return callback(properties, measurements, ...args);
+        })));
+    }
+
+    public registerEvent<T>(eventId: string, event: Event<T>, callback: (trackTelemetry: () => void, ...args: any[]) => any): void {
+        this.registerEventWithCustomTelemetry<T>(eventId, event, (trackTelemetry: () => void, _properties: TelemetryProperties, _measurements: TelemetryMeasurements, ...args: any[]) => callback(trackTelemetry, ...args));
+    }
+
+    public registerEventWithCustomTelemetry<T>(eventId: string, event: Event<T>, callback: (trackTelemetry: () => void, properties: TelemetryProperties, measurements: TelemetryMeasurements, ...args: any[]) => any): void {
+        this._extensionContext.subscriptions.push(event(this.wrapCallback(eventId, callback)));
+    }
+
+    private wrapCallback(callbackId: string, callback: (trackTelemetry: () => void, properties: TelemetryProperties, measurements: TelemetryMeasurements, ...args: any[]) => any): (...args: any[]) => Promise<any> {
+        return async (...args: any[]): Promise<any> => {
+            const start: number = Date.now();
+            let errorData: IParsedError | undefined;
+            const properties: TelemetryProperties = {};
+            const measurements: TelemetryMeasurements = {};
+            properties.result = 'Succeeded';
+            let sendTelemetry: boolean = false;
+
+            try {
+                await Promise.resolve(callback(() => { sendTelemetry = true; }, properties, measurements, ...args));
+            } catch (error) {
+                errorData = parseError(error);
+                // NOTE: Intentionally not using 'error instanceof UserCancelledError' because that doesn't work if multiple versions of the UI package are used in one extension
+                // See https://github.com/Microsoft/vscode-azuretools/issues/51 for more info
+                if (errorData.errorType === 'UserCancelledError') {
```

Comment:
How about adding errorData.isUserCancelledError, which will encourage the correct usage (and doesn't require comments everywhere in the calling code)?